### PR TITLE
Feature/cu 860rm5yuz

### PR DESCRIPTION
**Pull request by sarvar-swe**

### CODE REVIEW

**Comment by wahidustoz** [comment]
Task linked: [CU-860rm5yuz Fluent Configuration and Migration for entities](https://app.clickup.com/t/860rm5yuz)

File: Data/Configurations/QuizConfigurationMap.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+using X.Kwiz.Api.Entities;
+
+namespace X.Kwiz.Api.Data.Configurations;
+
+public class QuizConfigurationMap : IEntityTypeConfiguration<Quiz>
+{
+    public void Configure(EntityTypeBuilder<Quiz> builder)
+    {
+        builder.HasKey(b => b.Id);
+        builder.Property(b => b.Title).HasMaxLength(100).IsRequired();
```
Comment (wahidustoz) [inline Data/Configurations/QuizConfigurationMap.cs]: Title can be long

File: Data/Configurations/QuizConfigurationMap.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        builder.Property(b => b.Description).HasMaxLength(1024).IsRequired();
```
Comment (wahidustoz) [inline Data/Configurations/QuizConfigurationMap.cs:13]: Required emas

File: Data/Configurations/QuizConfigurationMap.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        builder.Property(b => b.OpeningDateTime).IsRequired();
+        builder.Property(b => b.ClosingDateTime).IsRequired();
+        builder.Property(b => b.Status).IsRequired();
+        builder.Property(b => b.IsPublic).IsRequired();
```
Comment (wahidustoz) [inline Data/Configurations/QuizConfigurationMap.cs]: Bool o'zi required boladi

File: Data/Configurations/QuizConfigurationMap.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+public class QuizConfigurationMap : IEntityTypeConfiguration<Quiz>
+{
+    public void Configure(EntityTypeBuilder<Quiz> builder)
+    {
+        builder.HasKey(b => b.Id);
+        builder.Property(b => b.Title).HasMaxLength(100).IsRequired();
+        builder.Property(b => b.Description).HasMaxLength(1024).IsRequired();
+        builder.Property(b => b.OpeningDateTime).IsRequired();
+        builder.Property(b => b.ClosingDateTime).IsRequired();
+        builder.Property(b => b.Status).IsRequired();
```
Comment (wahidustoz) [inline Data/Configurations/QuizConfigurationMap.cs]: Status enum by default required boladi

File: Data/Configurations/QuizConfigurationMap.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+        builder.Property(b => b.IsPublic).IsRequired();
+        builder.Property(b => b.Code).IsRequired();
```
Comment (wahidustoz) [inline Data/Configurations/QuizConfigurationMap.cs]: bu required emas.

File: Data/Configurations/QuizConfigurationMap.cs
Comment (wahidustoz) [inline Data/Configurations/QuizConfigurationMap.cs:1]: QuizQuestion bilan relationship qilmapsizku.
HasMany deb relationship qiilsh kerak

**Review by wahidustoz — COMMENTED** [review]
Value Typelar ga IsRequired degan fluent berish shart emas. 
Table lar o'rtasidagi Relationshiplarni ham qo'yib keting

File: Entities/Quiz.cs [hunk 6]
```diff
@@ -1,17 +1,18 @@
...
+    public DateTime UpdatedAt { get; set; }
     public DateTime OpeningDateTime { get; set; }
     public DateTime ClosingDateTime { get; set; }
     public EQuizStatus Status { get; set; }
-    public bool IsPublic { get; set; } 
+    public bool IsPublic { get; set; }
     public string[] Tags { get; set; }
     public string Code { get; set; }
     public List<QuizQuestion> Questions { get; set; }
+    public Guid QuizQuetionId { get; set; }
```
Comment (wahid-d) [inline Entities/Quiz.cs:17]:
```suggestion
    public Guid QuizQuestionId { get; set; }
```

Comment (wahidustoz) [inline Entities/Quiz.cs]:
```suggestion
```

File: Entities/QuizQuestion.cs [hunk 7]
```diff
@@ -4,6 +4,8 @@ public class QuizQuestion
     public Guid Id { get; set; }
     public int TimeLimitSeconds { get; set; }
     public string Content { get; set; }
-    public List<string> Options { get; set; }
+    public List<QuestionOption> Options { get; set; }
     public string[] Tags { get; set; }
+    public List<Quiz> Quizzes { get; set; }
+    public Guid QuizId { get; set; }
```
Comment (wahidustoz) [inline Entities/QuizQuestion.cs]:
```suggestion
```

File: Data/Configurations/QuizQuestionConfigurationMap.cs [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+        builder.Property(b => b.Content).HasMaxLength(1024).IsRequired();
+        
+        builder.HasMany(b => b.Options)
+            .WithOne(b => b.Question)
+            .HasForeignKey(b => b.QuestionId)
+            .IsRequired();
+        
+        builder.HasOne(b => b.SubmissionSelection)
+            .WithOne(b => b.Question)
+            .HasForeignKey<SubmissionSelection>(b => b.QuestionId);
```
Comment (wahidustoz) [inline Data/Configurations/QuizQuestionConfigurationMap.cs]: Mana bu navigation SubmissionSelection tarafdan bo'lishi kerak

File: Data/Configurations/QuizConfigurationMap.cs [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        builder.Property(b => b.Description).HasMaxLength(1024).IsRequired();
+
+        builder.HasMany(b => b.Questions)
+            .WithMany(b => b.Quizzes)
+            .UsingEntity(j => j.ToTable("QuizQuizQuestion"));
+        
+        builder.HasOne(b => b.Submission)
+            .WithOne(b => b.Quiz)
+            .HasForeignKey<Submission>(b => b.QuizId)
+            .IsRequired();
```
Comment (wahidustoz) [inline Data/Configurations/QuizConfigurationMap.cs]: manabu submission tarafdan bo'lishi kerak

File: Data/Configurations/SubmissionConfigurationMap.cs [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+public class SubmissionConfigurationMap : IEntityTypeConfiguration<Submission>
+{
+    public void Configure(EntityTypeBuilder<Submission> builder)
+    {
+        builder.HasKey(b => b.Id);
+
+        builder.HasOne(b => b.Quiz)
+            .WithOne(b => b.Submission)
+            .HasForeignKey<Quiz>(b => b.SubmissionId)
+            .IsRequired();
```
Comment (wahidustoz) [inline Data/Configurations/SubmissionConfigurationMap.cs]: Thiw is wrong. Wone to many bo'lishi kerak

File: Data/Configurations/SubmissionSelectionConfigurationMap.cs [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+namespace Kwiz.Api.Data.Configurations;
+
+public class SubmissionSelectionConfigurationMap : IEntityTypeConfiguration<SubmissionSelection>
+{
+    public void Configure(EntityTypeBuilder<SubmissionSelection> builder)
+    {
+        builder.HasKey(b => b.Id);
+
+        builder.HasOne(b => b.Question)
+            .WithOne(b => b.SubmissionSelection)
```
Comment (wahidustoz) [inline Data/Configurations/SubmissionSelectionConfigurationMap.cs]: bu ham xato. Qustion can have many submission selections

**Review by wahidustoz — APPROVED** [review]

**Pull request merged**